docs/docs/installation_guide/installation/pre-installation/oidc-providers.md

```diff
@@ -0,0 +1,127 @@
+
+Below are example procedures for creating OIDC providers in Authing, Keycloak and Okta for reference. You can use other OIDC providers as well.
+
+- (Authing Option) [Authing][authing] is an example of a third-party authentication provider. 
+- (Keycloak on AWS Option) [Keycloak on AWS][keycloak-solution] is a solution maintained by AWS and can serve as an authentication identity provider.
+- (Okta Option) [Okta][okta] is one trusted platform to secure every identity, from customers to your workforce.
+
+Follow the steps below to create an OIDC client, and obtain the `client_id` and `issuer`. 
+
+### (Authing Option) Authing.cn OIDC client
+
+1. Go to the [Authing console](https://console.authing.cn/console).
+2. Create a user pool if you don't have one.
+3. Select the user pool.
+4. On the left navigation bar, select **Self-built App** under **Applications**. 
+5. Click the **Create** button.
+6. Enter the **Application Name**, and **Subdomain**.
+7. Save the `App ID` (that is, `client_id`) and `Issuer` to a text file from Endpoint Information, which will be used later.
+    [![](../../../images/OIDC/endpoint-info.png)](../../../images/OIDC/endpoint-info.png)
+
+8. Update the `Login Callback URL` and `Logout Callback URL` to your IPC recorded domain name.
+    [![](../../../images/OIDC/authentication-configuration.png)](../../../images/OIDC/authentication-configuration.png)
+
+9. Set the Authorization Configuration.
+    [![](../../../images/OIDC/authorization-configuration.png)](../../../images/OIDC/authorization-configuration.png)
+
+You have successfully created an authing self-built application. 
+For more information, please visit [authing doc](https://docs.authing.cn/v2/en/)
+
+### (Keycloak on AWS Option) Keycloak OIDC client
+
+1. Deploy the Keycloak solution in AWS China Regions following [this guide](https://aws-samples.github.io/keycloak-on-aws/en/).
+
+2. Make sure you can log in to the Keycloak console.
+
+3. On the left navigation bar, select **Add realm**. Skip this step if you already have a realm. 
+
+4. Go to the realm setting page. Choose **Endpoints**, and then **OpenID Endpoint Configuration** from the list.
+
+    [![](../../../images/OIDC/keycloak-example-realm.jpg)](../../../images/OIDC/keycloak-example-realm.jpg)
+
+5. In the JSON file that opens up in your browser, record the **issuer** value which will be used later.
+
+    [![](../../../images/OIDC/OIDC-config.jpg)](../../../images/OIDC/OIDC-config.jpg)
+
+6. Go back to Keycloak console and select **Clients** on the left navigation bar, and choose **Create**.
+7. Enter a Client ID, which must contain 24 letters (case-insensitive) or numbers. Record the **Client ID** which will be used later.
+8. Change client settings. Enter `https://<Service Workbench Console Domain>` in **Valid Redirect URIs**，and enter `*` and `+` in **Web Origins**, as shown below.
```

Comment:
same here, how to configure if not using custom domain name

Reply:
I will add more description for the case that not using custom domain.